sql/src/main/java/io/crate/execution/engine/indexing/ShardDMLExecutor.java

```diff
@@ -126,6 +126,13 @@ private void addRowToRequest(TReq req, Row row) {
             BatchIterators.partition(batchIterator, bulkSize, requestFactory, this::addRowToRequest, r -> false);
         BinaryOperator<Long> combinePartialResult = (a, b) -> a + b;
         long initialResult = 0L;
+
+        // If the source batch iterator does not involve IO, mostly in-memory structures are used which we want to free
+        // as soon as possible. We do not want to throttle based on the targets node counter in such cases.
+        if (batchIterator.involvesIO() == false) {
+            shouldPause = ignored -> true;
```

Comment:
Probably just personal preference - but the global variable mutation wouldn't be necessary. Could instead introduce a local variable or use terse if syntax to parameterize the `BatchIteratorBackpressureExecutor`.

Feel free to ignore this nitpick if you prefer it the way it is.